jdaviz/core/helpers.py

```diff
@@ -982,3 +993,87 @@ def _next_subset_num(label_prefix, subset_groups):
                         max_i = i
 
     return max_i + 1
+
+
+class CubeConfigHelper(ImageConfigHelper):
+    """Base config helper class for cubes"""
+
+    _loaded_flux_cube = None
+    _loaded_uncert_cube = None
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def select_slice(self, value):
+        """
+        Select the slice closest to the provided value.
+
+        Parameters
+        ----------
+        value : float or int, optional
+            Slice value to select in units of the x-axis of the profile viewer.
+            The nearest slice will be selected if "snap to slice" is enabled in
+            the slice plugin.
+        """
+        msg = SliceSelectSliceMessage(value=value, sender=self)
+        self.app.hub.broadcast(msg)
+
+
+# TODO: We can remove this when specutils supports it, i.e.,
+#       https://github.com/astropy/specutils/issues/592 and
+#       https://github.com/astropy/specutils/pull/1009
+# NOTE: Cannot use custom_write decorator from specutils because
+#       that involves asking user to manually put something in
+#       their ~/.specutils directory.
+
+def jdaviz_cube_fitswriter(spectrum, file_name, **kwargs):
+    """This is a custom writer for Spectrum1D data cube.
+    This writer is specifically targetting data cube
+    generated from Cubeviz plugins (e.g., cube fitting)
+    with FITS WCS. It writes out data in the following format
+    (with MASK only exist when applicable)::
+
+        No.    Name      Ver    Type
+          0  PRIMARY       1 PrimaryHDU
+          1  SCI           1 ImageHDU (float32)
+          2  MASK          1 ImageHDU (uint16)
+
+    The FITS file generated by this writer does not need a
+    custom reader to be read back into Spectrum1D.
+
+    Examples
+    --------
+    To write out a Spectrum1D cube using this writer:
+
+    >>> spec.write("my_output.fits", format="jdaviz-cube", overwrite=True)  # doctest: +SKIP
+
+    """
+    pri_hdu = fits.PrimaryHDU()
+
+    flux = spectrum.flux
+    sci_hdu = fits.ImageHDU(flux.value.astype(np.float32))
+    sci_hdu.name = "SCI"
+    sci_hdu.header.update(spectrum.meta)
+    sci_hdu.header.update(spectrum.wcs.to_header())
+    sci_hdu.header['BUNIT'] = flux.unit.to_string(format='fits')
+
+    hlist = [pri_hdu, sci_hdu]
+
+    # https://specutils.readthedocs.io/en/latest/spectrum1d.html#including-masks
+    # Good: False or 0
+    # Bad: True or non-zero
+    if spectrum.mask is not None:
+        mask_hdu = fits.ImageHDU(spectrum.mask.astype(np.uint16))
+        mask_hdu.name = "MASK"
+        hlist.append(mask_hdu)
+
+    hdulist = fits.HDUList(hlist)
+    hdulist.writeto(file_name, **kwargs)
+
+
+if _astropy_has_priorities():
```

Comment:
is this still relevant with our pin of astropy?

Reply:
I checked and verified that `_astropy_has_priorities` checks for a `priority` kwarg in the `register_reader` function in v6.0.0. We currently pin `astropy >= 6`, so I think we can ditch this. Could you confirm @pllim?

Reply:
Wait... why are we reverting https://github.com/spacetelescope/jdaviz/pull/2094 ?